handler/core/src/test/java/io/knotx/fragments/handler/action/logging/InMemoryCacheActionFactoryLoggingTest.java

```diff
@@ -0,0 +1,281 @@
+            assertTrue(log.containsKey(CACHE_PASS_KEY));
+            assertEquals(FragmentResult.ERROR_TRANSITION,
+                log.getJsonObject(CACHE_PASS_KEY)
+                    .getString(TRANSITION_KEY));
+            testContext.completeNow();
+          });
+        });
+
+    assertTrue(testContext.awaitCompletion(5, TimeUnit.SECONDS));
+    if (testContext.failed()) {
+      throw testContext.causeOfFailure();
+    }
+  }
+
+  @DisplayName("Cache pass gets logged when no payload is added by doAction when log level is info")
+  @Test
+  void callActionThatDoesNotAddPayload(VertxTestContext testContext) throws Throwable {
+    Action doAction = (fragmentContext, resultHandler) -> Future
+        .succeededFuture(
+            new FragmentResult(fragmentContext.getFragment(), FragmentResult.SUCCESS_TRANSITION))
+        .setHandler(resultHandler);
+
+    Action tested = new InMemoryCacheActionFactory()
+        .create(ACTION_ALIAS, ACTION_CONFIG.copy().put(LOG_LEVEL_KEY, "info"), null, doAction);
+
+    // when
+    tested.apply(new FragmentContext(firstFragment, new ClientRequest()),
+        result -> {
+          // then
+          testContext.verify(() -> {
+            JsonObject log = result.result().getNodeLog().getJsonObject(LOGS_KEY);
+            assertTrue(log.containsKey(CACHE_PASS_KEY));
+            testContext.completeNow();
+          });
+        });
+
+    assertTrue(testContext.awaitCompletion(5, TimeUnit.SECONDS));
+    if (testContext.failed()) {
+      throw testContext.causeOfFailure();
+    }
+  }
+
+
+  @DisplayName("Cache operations are not logged when log level is not set")
+  @Test
+  void callActionWithSuccessTransitionWithLogLevelNotSet(VertxTestContext testContext)
+      throws Throwable {
+    // given
+    Action doAction = (fragmentContext, resultHandler) -> Future
+        .succeededFuture(new FragmentResult(firstFragment, FragmentResult.SUCCESS_TRANSITION))
+        .setHandler(resultHandler);
+
+    Action tested = new InMemoryCacheActionFactory()
+        .create(ACTION_ALIAS, ACTION_CONFIG, null, doAction);
+
+    // when
+    tested.apply(new FragmentContext(firstFragment, new ClientRequest()),
+        result -> {
+          // then
+          testContext.verify(() -> {
+            JsonObject log = result.result().getNodeLog().getJsonObject(LOGS_KEY);
+            assertFalse(log.containsKey(CACHE_LOOKUP_KEY));
+            assertFalse(log.containsKey(CACHE_HIT_KEY));
+            assertFalse(log.containsKey(CACHE_MISS_KEY));
+            assertFalse(log.containsKey(CACHE_PASS_KEY));
+            testContext.completeNow();
+          });
+        });
+
+    assertTrue(testContext.awaitCompletion(5, TimeUnit.SECONDS));
+    if (testContext.failed()) {
+      throw testContext.causeOfFailure();
+    }
+  }
+
+  @DisplayName("Cache operations are not logged when log level is error")
+  @Test
+  void callActionWithSuccessTransitionWithLogLevelError(VertxTestContext testContext)
```

Comment:
This test is duplicated, can we implement parameterized test? (see https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests)

Examples: https://github.com/Knotx/knotx-fragments/blob/master/handler/core/src/test/java/io/knotx/fragments/handler/action/cb/CircuitBreakerActionFactoryTest.java#L95

Reply:
Good idea, will update the PR